src/jit/lower.cpp

```diff
@@ -5724,8 +5722,9 @@ void Lowering::ContainCheckRet(GenTreeOp* ret)
 void Lowering::ContainCheckJTrue(GenTreeOp* node)
 {
     // The compare does not need to be generated into a register.
-    GenTree* cmp                   = node->gtGetOp1();
-    cmp->gtLsraInfo.isNoRegCompare = true;
+    GenTree* cmp = node->gtGetOp1();
+    cmp->gtType  = TYP_VOID;
+    cmp->gtFlags |= GTF_SET_FLAGS;
```

Comment:
I suppose this doesn't hurt but I don't think we ever set `GTF_SET_FLAGS` on a relop.

Reply:
Right; but once it has `TYP_VOID` and is a non-value, we have to have a way to identify whether or not its result is used.

Reply:
Ah, interesting. So JTRUE's `gtOp1` still points to the relop but since the relop is not a value anymore that's not enough to keep it alive. It basically turns relop+JTRUE into a slightly bizarre variation of CMP+JCC.

It looks like proper lowering to JCC is ultimately the "right way". But this will do for now :)